code/modules/mob/living/simple_animal/friendly/corgi.dm

```diff
@@ -462,18 +461,26 @@
 					to_chat(user, "<span class='warning'>There is nothing to remove from its [remove_from].</span>")
 				return
 
+#define IDLE 0
+#define BEGIN_FOOD_HUNTING 1
+#define FOOD_HUNTING 2
+#define BEGIN_POINTER_FOLLOWING 3
+#define POINTER_FOLLOWING 4
+
 //IAN! SQUEEEEEEEEE~
 /mob/living/simple_animal/corgi/Ian
 	name = "Ian"
 	real_name = "Ian"	//Intended to hold the name without altering it.
 	gender = MALE
 	desc = "It's a corgi."
-	var/turns_since_scan = 0
 	var/obj/movement_target
+	var/mob/pointer_caller
+	var/mob/master //Obtained randomly when petting him. Can be overriden.
```

Comment:
Don't forget to set this to `null` in `Destroy()`.
You also want the master to be `null`'d on Ian's end if the mob is destroyed in another way.